src/security.rs

```
@@ -0,0 +1,81 @@
+use crate::routes::ServerError;
+use futures_util::future::LocalBoxFuture;
+use std::future::{ready, Ready};
+
+use actix_web::{
+    body::EitherBody,
+    dev::{self, Service, ServiceRequest, ServiceResponse, Transform},
+    Error,
+};
+
+use ipnet::IpNet;
+use std::net::IpAddr;
+
+pub struct ExportAuthentification;
+
+impl<S, B> Transform<S, ServiceRequest> for ExportAuthentification
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = ServiceResponse<EitherBody<B>>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = ExportAuthentificationMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        let raw_address_ranges: Vec<String> = serde_json::from_str(
+            &std::env::var("PROMETHEUS_ALLOWED_IPS").unwrap_or("[\"10.13.37.0/24\"]".to_string()),
```

No defaults. Not configured -> all requests to the endpoint gracefully HTTP error out

---

So u just want an expect(...) ?

---

also possible, like if we want to just configure nothing or hard fail (expect) on missing config - up to you